pkg/userdata/ubuntu/userdata.go

```diff
@@ -62,7 +62,7 @@ func (p Provider) UserData(
 		return "", fmt.Errorf("failed to get cloud config: %v", err)
 	}
 
-	pconfig, err := providerconfig.GetConfig(spec.ProviderConfig)
+	pconfig, err := providerconfig.GetConfig(spec.ProviderSpec)
 	if err != nil {
 		return "", fmt.Errorf("failed to get provider config: %v", err)
```

Comment:
Should be replaced with `failed to get provider spec`

Would be great if you could grep the whole code for `provider config` and replace it